src/files.cpp

```diff
@@ -0,0 +1,89 @@
+#include <fstream>
+#include <sstream>
+#include "files.h"
+
+
+FileManager::Exception::Exception(std::string msg) : msg{std::move(msg)} {}
+
+const char* FileManager::Exception::what() const noexcept {
+    return this->msg.c_str();
+}
+
+std::vector<char> FileManager::read_file(const fs::path& path) {
+    std::vector<char> result;
+    std::ifstream f{path};
+    f.seekg(0, std::ios::end);
+    long size = f.tellg();
+    if (size < 0) {
+        std::stringstream exc_msg_ss;
+        exc_msg_ss << "error while opening file '" << path.string() << "'";
+        throw FileManager::Exception(exc_msg_ss.str());
+    }
+    result.resize(static_cast<unsigned long>(size));
```

Comment:
Why not cast to `std::size_t`?

```suggestion
    result.resize(static_cast<std::size_t>(size));
```